### PR TITLE
Allow setting different regions to generated AWS resources

### LINKED ISSUE #150: Allow different regions in AWS resource creation functions
# Description

Part of #133 was to make the necessary modifications to be able to create resources across multiple regions due to the `test_regions` tests. Currently, the framework generates a client to connect to AWS using the `us-east-1` region, therefore if we want to generate and fetch events from other regions the region would need to be configurable, according to the test case being executed.

### PR DESCRIPTION
**Pull request by fdalmaup**
# Description

Closes #150. 
- Modifies S3 data generation methods and functions to generate resources for any specified region with the possibility to include `prefix` or `suffix`.
- Removes the `boto3` client instantiations from the framework, making it the responsibility of the executed tests to instantiate and pass them accordingly.
- Fixes related AWS database functions that were not correctly used after the rebase of the epic branch.

## Tests

<details><summary>Current working tests execution</summary>

```console
root@vagrant:/# pytest -x /wazuh/tests/integration/test_aws/test_log_groups.py 
=================================================================== test session starts ====================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.1.1, html-3.1.1
collected 2 items                                                                                                                                          

wazuh/tests/integration/test_aws/test_log_groups.py ..                                                                                               [100%]

==================================================================== 2 passed in 53.73s ====================================================================
root@vagrant:/# pytest -x /wazuh/tests/integration/test_aws/test_discard_regex.py
=================================================================== test session starts ====================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.1.1, html-3.1.1
collected 17 items                                                                                                                                         

wazuh/tests/integration/test_aws/test_discard_regex.py .................                                                                             [100%]

============================================================== 17 passed in 416.51s (0:06:56) ==============================================================
root@vagrant:/# pytest -x /wazuh/tests/integration/test_aws/test_basic.py 
=================================================================== test session starts ====================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.1.1, html-3.1.1
collected 16 items                                                                                                                                         

wazuh/tests/integration/test_aws/test_basic.py ................                                                                                      [100%]

============================================================== 16 passed in 310.19s (0:05:10) ==============================================================
root@vagrant:/# pytest -x /wazuh/tests/integration/test_aws/test_custom_bucket.py 
=================================================================== test session starts ====================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.1.1, html-3.1.1
collected 2 items                                                                                                                                          

wazuh/tests/integration/test_aws/test_custom_bucket.py ..                                                                                            [100%]

=============================================================== 2 passed in 80.79s (0:01:20) ===============================================================
```

</details>

### CODE REVIEW

File: src/wazuh_testing/modules/aws/data_generator.py [hunk 1]
```diff
@@ -1167,13 +1192,14 @@ def get_data_sample(self):
 }
 
 
-def get_data_generator(bucket_type, bucket_name, creation_date) -> DataGenerator:
+def get_data_generator(bucket_type, bucket_name, creation_date, region) -> DataGenerator:
```
Comment (GGP1) [inline src/wazuh_testing/modules/aws/data_generator.py:1195]: Now that the region is an input, could we validate that it is valid and fail early? Something like

```py
from boto3.session import Session
...

session = Session()
service_regions = session.get_available_regions('<service>')
if region not in service_regions:
    raise Exception(f"Invalid region, available regions are: {', '.join(service_regions)}")
```

File: src/wazuh_testing/modules/aws/utils.py [hunk 2]
```diff
@@ -155,30 +146,32 @@ def upload_bucket_file(bucket_name: str, data: str, key: str):
...
 
 
-def delete_bucket_file(filename: str, bucket_name: str):
+def delete_bucket_file(filename: str, bucket_name: str, client):
     """Delete a given file from the bucket.
 
     Args:
         filename (str): Full filename to delete.
         bucket_name (str): Bucket that contains the file.
+        client (boto3.resources.base.ServiceResource): S3 client used to delete tbe file from the bucket.
```
Comment (GGP1) [inline src/wazuh_testing/modules/aws/utils.py:155]:
```suggestion
        client (boto3.resources.base.ServiceResource): S3 client used to delete the file from the bucket.
```

File: src/wazuh_testing/modules/aws/utils.py [hunk 3]
```diff
@@ -190,17 +183,18 @@ def get_last_file_key(bucket_type: str, bucket_name: str, execution_datetime: da
...
     Args:
         vpc_name (str): Name to tag the created VPC.
+        client (Service client instance): EC2 client to create a VPC.
         
     Returns:
         vpc_id (str): ID of the VPC created.
     """
     try:
-        vpc = ec2.create_vpc(CidrBlock= '10.0.0.0/16',
+        vpc = client.create_vpc(CidrBlock= '10.0.0.0/16',
```
Comment (GGP1) [inline src/wazuh_testing/modules/aws/utils.py:197]:
```suggestion
        vpc = client.create_vpc(CidrBlock='10.0.0.0/16',
```

File: src/wazuh_testing/utils/db_queries/aws_db.py [hunk 4]
```diff
@@ -131,8 +131,8 @@ def get_s3_db_row(table_name) -> S3CloudTrailRow:
     """
     row_type = _get_s3_row_type(table_name)
     query = SELECT_QUERY_TEMPLATE.format(table_name=table_name)
-    row = get_fetch_one_query_result(S3_CLOUDTRAIL_DB_PATH, query)
-
+    row = get_sqlite_fetch_one_query_result(S3_CLOUDTRAIL_DB_PATH, query)[0]
+
```
Comment (GGP1) [inline src/wazuh_testing/utils/db_queries/aws_db.py:135]:
```suggestion

```

File: src/wazuh_testing/utils/database.py [hunk 5]
```diff
@@ -134,6 +134,25 @@ def get_sqlite_query_result(db_path: str, query: str) -> List[str]:
         services.control_service('start', daemon=WAZUH_DB_DAEMON)
 
 
+def get_sqlite_fetch_one_query_result(db_path: str, query: str) -> tuple:
```
Comment (GGP1) [inline src/wazuh_testing/utils/database.py:137]: [db.execute_fetch_one_query](https://github.com/wazuh/qa-integration-framework/blob/1321b3bd96078745949ea2b55c3b5989f5e32387/src/wazuh_testing/tools/db_administrator.py#L41) is returning a list

```suggestion
def get_sqlite_fetch_one_query_result(db_path: str, query: str) -> list:
```

File: src/wazuh_testing/utils/database.py [hunk 6]
```diff
@@ -134,6 +134,25 @@ def get_sqlite_query_result(db_path: str, query: str) -> List[str]:
...
+    """Execute a query expecting only one result in a given database and return the result.
+    Args:
+        db_path (str): Path where is located the DB.
+        query (str): SQL query. e.g(SELECT * ..).
+    Returns:
+        result (tuple): Query result row.
```
Comment (GGP1) [inline src/wazuh_testing/utils/database.py:143]:
```suggestion
        result (list): Query result row.
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: src/wazuh_testing/utils/database.py (hunk 5 above)
Comment (fdalmaup) [reply src/wazuh_testing/utils/database.py:137]: You are right, but the [documentation](https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.fetchone) mentions that a `tuple` is returned

File: src/wazuh_testing/modules/aws/data_generator.py (hunk 1 above)
Comment (fdalmaup) [reply src/wazuh_testing/modules/aws/data_generator.py:1195]: Done

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by Selutario — APPROVED** [review]
It's looking great.

**Pull request merged**